client/components/theme-tier/theme-tier-badge/index.js

```diff
@@ -23,49 +24,77 @@ export default function ThemeTierBadge( {
 	showUpgradeBadge = true,
 	themeId,
 	showPartnerPrice = false,
+	hideBackgroundOnUpgrade = false,
 } ) {
 	const siteId = useSelector( getSelectedSiteId );
-	const themeType = useSelector( ( state ) => getThemeType( state, themeId ) );
-	const isLegacyPremiumPurchased = useSelector( ( state ) =>
-		isThemePurchased( state, themeId, siteId )
+	const themeType = useSelector(
+		( state ) => getThemeType( state, themeId ),
+		( prev, next ) => prev === next
 	);
 	const themeTier = useThemeTierForTheme( themeId );
 	const isThemeAllowed = useIsThemeAllowedOnSite( siteId, themeId );
-	const isUpdatedBadgeDesign = useIsUpdatedBadgeDesign();
-	const getBadge = () => {
-		if ( isUpdatedBadgeDesign && 'free' === themeTier?.slug ) {
+
+	const getBadge = useMemo( () => {
```

Comment:
```suggestion
	const badge = useMemo( () => {
```

I'd rename the function, since having it as `getBadge` suggests that it's a getter that needs to be called even after the `useMemo` in order to get the value of the badge. Wdyt?

Reply:
Agree! Thanks